lib/model.js

```diff
@@ -18,129 +18,98 @@ var indexModelProps = {
-      fn: function() {
-        return this.name === '_id_' || !!this.extra.unique;
-      }
-    },
-    sparse: {
-      deps: ['extra'],
-      fn: function() {
-        return !!this.extra.sparse;
-      }
-    },
-    ttl: {
-      deps: ['extra'],
-      fn: function() {
-        return !!this.extra.expireAfterSeconds;
-      }
-    },
-    hashed: {
-      deps: ['key'],
-      fn: function() {
-        return _.values(this.key).indexOf('hashed') > -1;
-      }
-    },
-    geo: {
-      deps: ['extra', 'key'],
-      fn: function() {
-        return !!this.extra['2dsphereIndexVersion'] ||
-          _.values(this.key).indexOf('2d') > -1 ||
-          _.values(this.key).indexOf('geoHaystack') > -1;
-      }
-    },
-    compound: {
-      deps: ['key'],
-      fn: function() {
-        return _.keys(this.key).length > 1;
-      }
-    },
-    single: {
-      deps: ['compound'],
-      fn: function() {
-        return !this.compound;
-      }
-    },
-    partial: {
-      deps: ['extra'],
-      fn: function() {
-        return !!this.extra.partialFilterExpression;
-      }
-    },
-    text: {
-      deps: ['extra'],
-      fn: function() {
-        return !!this.extra.textIndexVersion;
-      }
-    },
-    collation: {
-      deps: ['extra'],
-      fn: function() {
-        return !!this.extra.collation;
-      }
-    },
-    type: {
-      deps: ['geo', 'hashed', 'text'],
-      fn: function() {
-        if (this.geo) {
-          return 'geospatial';
-        }
-        if (this.hashed) {
-          return 'hashed';
-        }
-        if (this.text) {
-          return 'text';
-        }
-        return 'regular';
-      }
-    },
-    cardinality: {
-      deps: ['single'],
-      fn: function() {
-        return this.single ? 'single' : 'compound';
-      }
-    },
-    properties: {
-      deps: ['unique', 'sparse', 'partial', 'ttl', 'collation'],
-      fn: function() {
-        var model = this;
-        var props = ['unique', 'sparse', 'partial', 'ttl', 'collation'];
-        return _.filter(props, function(prop) {
-          return !!model[prop];
-        }).sort(function(a, b) {
-          var order = {
-            'unique': 1,
-            'sparse': 2,
-            'partial': 3,
-            'ttl': 4,
-            'collation': 5
-          };
-          return order[a] - order[b];
-        });
-      }
+class IndexModel {
+  /**
+   * Instantiate the warning.
+   *
+   * @param {String} ns - The namespace.
+   * @param {Object} key - The index key.
+   * @param {String} name - The index name.
+   * @param {Number} size - The size of the index.
+   * @param {Number} usageCount - The number of times the index has been used.
+   * @param {Date} usageSince - Last time used.
+   * @param {String} usageHost - The ost.
+   * @param {Number} version - The version.
+   * @param {Object} extra - Extra info.
+   */
+  constructor(ns, key, name, size, usageCount, usageSince, usageHost, version, extra) {
```

Comment:
better use an `options` object instead of all variables separately. Easier to extend and easier to use ("what was the 5th argument again?").